processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -846,6 +859,19 @@ func (c *WatchClient) addOrUpdateReplicaSet(replicaset *apps_v1.ReplicaSet) {
 	c.m.Unlock()
 }
 
+// This function removes all data from the Pod except what is required by extraction rules
+func removeUnnecessaryReplicaSetData(replicaset *apps_v1.ReplicaSet) *apps_v1.ReplicaSet {
+	transformedReplicaset := apps_v1.ReplicaSet{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      replicaset.GetName(),
+			Namespace: replicaset.GetNamespace(),
+			UID:       replicaset.GetUID(),
+		},
+	}
+	transformedReplicaset.SetOwnerReferences(replicaset.GetOwnerReferences())
```

Comment:
It might be helpful to leave a comment here about what's going on. 

Admittedly I'm not very familiar with this area, but it took a while to understand that this line was the filtering logic, and the thought is that we're just keeping ReplicaSets that are owned by another object.

Reply:
Okay, maybe I confused myself here. By unnecessary data we're referring to fields and metadata of the original `replicaset` that we can safely remove? Is that the idea?

Reply:
Yes. If you'd like a more detailed expose on why this is necessary, #23272 might be clearer. The TLDR is that since we're using informers, we keep a local cache of resources from the API Server, and this function intercepts them before they're stored and strips all the data we don't need. In practice it allocates a new object and copies the fields we need over, because that's way simpler than the reverse.